mmv1/third_party/terraform/resources/resource_apigee_shared_flow.go

```diff
@@ -0,0 +1,439 @@
+	}
+	log.Printf("[DEBUG] resourceApigeeSharedFlowRead sendRequest completed")
+	previousLastModifiedAt := getApigeeSharedFlowLastModifiedAt(d)
+	if err := d.Set("meta_data", flattenApigeeSharedFlowMetaData(res["metaData"], d, config)); err != nil {
+		return fmt.Errorf("Error reading SharedFlow: %s", err)
+	}
+	currentLastModifiedAt := getApigeeSharedFlowLastModifiedAt(d)
+	if err := d.Set("name", flattenApigeeSharedFlowName(res["name"], d, config)); err != nil {
+		return fmt.Errorf("Error reading SharedFlow: %s", err)
+	}
+	if err := d.Set("revision", flattenApigeeSharedFlowRevision(res["revision"], d, config)); err != nil {
+		return fmt.Errorf("Error reading SharedFlow: %s", err)
+	}
+	if err := d.Set("latest_revision_id", flattenApigeeSharedFlowLatestRevisionId(res["latestRevisionId"], d, config)); err != nil {
+		return fmt.Errorf("Error reading SharedFlow: %s", err)
+	}
+
+	//setting hash to suggest update
+	if previousLastModifiedAt != currentLastModifiedAt {
+		log.Printf("yoyo LastModifiedAt previous: %s, current: %s", previousLastModifiedAt, currentLastModifiedAt)
+		d.Set("md5hash", "UNKNOWN")
+		d.Set("detect_md5hash", "UNKNOWN")
+	}
+	return nil
+}
+
+func getApigeeSharedFlowLastModifiedAt(d *schema.ResourceData) string {
+
+	metaDataRaw := d.Get("meta_data").([]interface{})
+	if len(metaDataRaw) != 1 {
+		//in Terraform Schema, a nest in object is implemented as an array of length one, even if it's technically an object
+		return "UNKNOWN"
+	}
+	metaData := metaDataRaw[0].(map[string]interface{})
+	if metaData == nil {
+		return "UNKNOWN"
+	}
+	lastModifiedAt := metaData["last_modified_at"].(string)
+	if lastModifiedAt == "" {
+		return "UNKNOWN"
+	}
+	return lastModifiedAt
+}
+
+func resourceApigeeSharedFlowDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] resourceApigeeSharedFlowDelete")
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	billingProject := ""
+
+	url, err := replaceVars(d, config, "{{ApigeeBasePath}}organizations/{{org_id}}/sharedflows/{{name}}")
+	if err != nil {
+		return err
+	}
+
+	var obj map[string]interface{}
+	log.Printf("[DEBUG] Deleting SharedFlow %q", d.Id())
+
+	// err == nil indicates that the billing_project value was found
+	if bp, err := getBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+
+	res, err := sendRequestWithTimeout(config, "DELETE", billingProject, url, userAgent, obj, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return handleNotFoundError(err, d, "SharedFlow")
+	}
+
+	log.Printf("[DEBUG] Finished deleting SharedFlow %q: %#v", d.Id(), res)
+	return nil
+}
+
+func resourceApigeeSharedFlowImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
+	if err := parseImportId([]string{
+		"organizations/(?P<org_id>[^/]+)/sharedflows/(?P<name>[^/]+)/(?P<name>[^/]+)",
```

Comment:
i'm not entirely sure what's going on here without stepping through it, but the error i'm seeing is that the url being created in the read function is `https://apigee.googleapis.com/v1/organizations/organizations/sharedflows/sharedflows`.

I think `name` is being duplicated here, right? we just want one `name` after `sharedflows`? and probably in the regex on the next line as well.

Reply:
Ah for some reason my commit on this file was lost, the regex should have been 
```
"organizations/(?P<org_id>[^/]+)/sharedflows/(?P<name>[^/]+)",
		"(?P<org_id>[^/]+)/(?P<name>[^/]+)",
```